### PR TITLE
cli: avoid glog flags inject into tidb cli

### LINKED ISSUE #47204: glog flag in the tidb cli
## Bug Report

Please answer these questions before submitting your issue. Thanks!

glog's flag should not be in the tidb cli.

### 1. Minimal reproduce step (Required)

tidb-server --help

### 2. What did you expect to see? (Required)

```
  -log-file string
        log file path
  -log-slow-query string
        slow query file path
```

### 3. What did you see instead (Required)

```
  -log-file string
        log file path
  -log-slow-query string
        slow query file path
  -log_backtrace_at value
        when logging hits line file:N, emit a stack trace
  -log_dir string
        If non-empty, write log files in this directory
  -log_link string
        If non-empty, add symbolic links in this directory to the log files
  -logbuflevel int
        Buffer log messages logged at this level or lower (-1 means don't buffer; 0 means buffer INFO only; ...). Has limited applicability on non-prod platforms.
  -logtostderr
        log to standard error instead of files

```

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #47204

Problem Summary:

### What is changed and how it works?

In the initial version, the initialization process would include glog. Now, with this update, we have made a few changes. After the main function starts, the ```initflag``` function will be cleared first, and then the program will continue to execute.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

```
tidb-server --help

  -log-file string
        log file path
  -log-slow-query string
        slow query file path

```

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tiancaiamao — APPROVED** [review]

File: tidb-server/main.go [hunk 1]
```diff
@@ -134,59 +134,115 @@ const (
...
+	initializeInsecure          *bool
+	initializeSQLFile           *string
+	disconnectOnExpiredPassword *bool
+	keyspaceName                *string
+	serviceScope                *string
+	help                        *bool
+)
+
+func initflag() {
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
```
Comment (xhebox) [inline tidb-server/main.go:189]: What about just `fpset := flag.NewFlagSet`?

Comment (hawkingrei) [reply tidb-server/main.go:189]: In fact, the global one have been polluted by the glog. so we need to clean it.

Comment (hawkingrei) [reply tidb-server/main.go:189]: I try it. it is unavailable.

Comment (hawkingrei) [reply tidb-server/main.go]: I try it again. it works.

**Review by xhebox — APPROVED** [review]

**Pull request merged**